### ISSUE TITLE
Launching docker container in interactive mode

### ISSUE THREAD

**Issue by betatim** [jupyter/repo2docker#599]
I was expecting `repo2docker https://github.com/binder-examples/conda-freeze /bin/bash` to give me a shell inside the container. Instead it builds the image and then exits straight away.

I think this is because we would have to do the equivalent of `docker run -it <image> /bin/bash` but don't. Should we add a `--interactive` CLI flag that turns on this behaviour?

**Comment by choldgraf** [comment]
so you're thinking that it'd send you to shell *after* the jupyter server starts?

**Comment by betatim** [comment]
No. If I specify a command `repo2docker ..stuff.. /bin/bash` I expect it to not run the jupyter server but only the command I specified. This works except it doesn't connect me to the "end" of the bash session.

"Interactive mode" is the difference between `docker run --rm ubuntu:latest /bin/bash` vs `docker run -it --rm ubuntu:latest /bin/bash`.

**Comment by yuvipanda** [comment]
I *think* implementing the equivalent of '--interactive' to docker run without using the docker binary directly is not easy. See https://github.com/docker/docker-py/issues/247.

**Comment by betatim** [comment]
Maybe a quick win is to have `repo2docker --interactive ...` (or some such) print out the `docker run` command to use as a form of syntactic sugar. Have you tried https://github.com/d11wtq/dockerpty?

**Comment by yuvipanda** [comment]
@betatim I like the idea of us printing the command.
